internal/entities/currency.go

```diff
@@ -1,27 +1,30 @@
 package entities
 
+import "strings"
+
 type Currency string
 
 const (
 	AmericanDollar   Currency = "USD"
 	UkrainianHryvnia Currency = "UAH"
 )
 
-var SupportedCurrencies = []Currency{AmericanDollar, UkrainianHryvnia}
+var SupportedCurrencies = map[Currency]struct{}{
```

Comment:
As a domain entity better not to expose it to entire world. You can do smth like getter to get all available currencies.
For example 
```go
func GetAllCurencies() []string {
     return []string{
          ...
    }
}
```

Reply:
Made array with all supported currencies private